drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -73,4 +80,57 @@ def factory_name
   def schema
     raise 'Implement in sub-class'
   end
+
+  def process_custom_field(field, value)
```

Comment:
Good point. There shouldn't be too many custom fields- maybe a handful.

This code doesn't run in a transaction, since it doesn't mutate the db at all. It is just assigning attributes. Why do you say it should be in a transaction? The actual save runs in a transaction of course.